### PR TITLE
Add mismapping detection calibration to mpmap

### PR DESCRIPTION
**Pull request by jeizenga**
I implemented a strategy to calibrate the MultipathMapper's mismapping detection to a graph and to read length. I think this method will be more robust to high-error reads. It does take a little bit of start up cost (not scaling with job size and scaling very sublinearly with graph size) that might show up disproportionately on the small tests on Jenkins. We'll see I guess.

### CODE REVIEW

**Comment by glennhickey** [comment]
That looks like it should work.  Is this causing the failed report here?

http://cgl-pipeline-inputs.s3.amazonaws.com/vg_cgl/vg_ci/jenkins_reports/commit/b8c0150f537ccb63b9fdf9ff854e8f1178b10f7c/index.html

**Comment by jeizenga** [comment]
I think so. It's all I changed between f192908 (which passed tests) and a390f39 (which blew up). The -B option works for me locally too.

**Comment by glennhickey** [comment]
I'll take a look locally here.  Debugging this is aggravated by the fact
that the mapping logs somehow moved from stderr to stdout in the simulation
tests.

**Comment by glennhickey** [comment]
I can reproduce locally with the small test (after removing its skip
decorator)

jenkins/jenkins.sh -l -r -k  -t
jenkins/vgci.py::VGCITest::test_sim_brca2_snp1kg_mpmap

get a bunch of the same errors.  The logs in ./test-report.xml show mpmap
seems be getting called right:

Run: vg mpmap -B -A -G reads_chunk_0_0.gam -t 8 -i -x graph.vg.xg -g
graph.vg.gcsa
2017-12-04 15:56:39,564 - toil-rt - INFO - Successfully ran vg mpmap -B -A
-G reads_chunk_0_0.gam -t 8 -i -x graph.vg.xg -g graph.vg.gcsa in
37.3022871017 seconds.

But the output files look like nonsense

vg view -a
vgci-work/outstore-sim-BRCA2-snp1kg-mpmap/aligned-snp1kg-mp_default.gam |
more

{"sequence":
"AAATCAGGCCTTCTTACTTTATTTGCTGGAGATTTTTCTGTGTTTTCTGCGAGTACAAAAGAGGGCCACTTTCAACAGACATTCATCAAAATGAAAAATACTGTTGAGGTAAGGTTACTTTTCAGTATCTCCACACATTTTGGTATTTCT",
"name
": "936424a92ebd3ca3_1", "query_position": 60}
{"sequence":
"TTATATAATGTATATATAGATTTAAGCAGAAATGCCGTACATGATCTAGGGAAAATAATATCAGTGTCTCCTATCTGGGATTAAGTCAAGTCAAAGAAATATTAATCTGTATAAAACACTAGAAATGGAGTTTAAATGTAATCCTGTAGT",
"name
": "936424a92ebd3ca3_2", "query_position": 60}
etc. etc.

Could it be the High Sierra root bug?

**Comment by jeizenga** [comment]
No idea

**Comment by glennhickey** [comment]
Took another look.  A quirk of the interface of specifying those overrides
is that it overrides the default mpmap_opts (['-S']).  The toil-vg sanity
check for this is bypassed when not running on the command line so there's
no useful error message.

tldr try:

mpmap_opts = ['-S', '-B']

instead of

mpmap_opts = ['-B']

**Comment by jeizenga** [comment]
Jenkins, test this please.

**Comment by glennhickey** [comment]
Looks like Jenkins is getting rebooted: "Jenkins is going to shut down. No
further builds will be performed. "

**Comment by jeizenga** [comment]
Jenkins, test this please.
